libbeat/cmd/instance/beat.go

```diff
@@ -855,6 +862,30 @@ func (b *Beat) loadDashboards(ctx context.Context, force bool) error {
 	return nil
 }
 
+// warnAboutElasticsearchVersion registers a global callback to warn users once
+// if they try to connect to an older Elasticsearch version than the Beat version.
+func (b *Beat) warnAboutElasticsearchVersion() {
+	if b.Config.Output.Name() != "elasticsearch" {
+		return
+	}
+
+	elasticsearch.RegisterGlobalCallback(func(conn *eslegclient.Connection) error {
+		warnAboutOldES.Do(func() {
+			esVersion := conn.GetVersion()
+			beatVersion, err := common.NewVersion(b.Info.Version)
+			if err != nil {
+				logp.Debug("beat", "Failed to determine Beat version to compare it with ES version: %+v", err)
```

Comment:
At https://github.com/elastic/beats/pull/29683/files#diff-7892d09d6cd53dcdb72155590e61e694ebea064f17965984cfc22286f02f3e55R872 we don't log this, should we?